### DISCUSSION TITLE
Hi, How to draw text with font   load from file ?

### DISCUSSION THREAD

**Discussion by toplinuxsir** [rust-skia/rust-skia#658] [Q&A] [answered]
Hi, How to get font from font file(ttf) ? any example ?
Thanks!

**Comment by tronical** [comment, marked as answer]
In principle you need to get a `skia_safe::TypeFace` from `TypeFace::from_data` (after reading the file). Once you have the `skia_safe::TypeFace` you can create a `skia_safe::Font` with `Font::from_typeface`, and use that with with a `skia_safe::TextBlob`.

If you're using the `textlayout` module, then in your `FontCollection` you may want to install a `TypefaceFontProvider` and use its `register_typeface` function to make your custom font known to the provide, collection and finally `ParagraphBuilder`.

**Comment by toplinuxsir** [comment]
@tronical Thanks! It works!
